modules/genomic_browser/ajax/genomic_file_upload.php

```diff
@@ -382,45 +437,46 @@ function (&$item, $key, $prefix) {
  */
 function createCandidateFileRelations(&$fileToUpload)
 {
-    $config           = NDB_Config::singleton();
-    $genomic_data_dir = $config->getSetting('GenomicDataPath');
     $DB =& Database::singleton();
 
     $f = fopen(
-        $genomic_data_dir . 'genomic_uploader/'
-            . $fileToUpload->file_name,
+        $fileToUpload->tmp_name,
         "r"
     );
 
     if ($f) {
-
-        $stmt_prefix = 'INSERT IGNORE INTO genomic_candidate_files_rel
-            (CandID, GenomicFileID) VALUES ';
-        $stmt        = '';
-
-        $line = fgets($f);
-        fclose($f);
-
-        $headers = explode(',', $line);
-        array_shift($headers);
-
-        $rows = array();
-        foreach ($headers as $pscid) {
-            $pscid = trim($pscid);
-            $row   = "(
-               (select CandID from candidate where PSCID = '$pscid'),
-               $fileToUpload->GenomicFileID
-            )";
-            array_push($rows, $row);
-        }
-        $stmt .= join(',', $rows);
-
         try {
+            $line = fgets($f);
+            fclose($f);
+
+            $headers = explode(',', $line);
+            array_shift($headers);
+
+            $stmt = $DB->prepare(
+                "
+                INSERT IGNORE INTO
+                    genomic_candidate_files_rel (CandID, GenomicFileID)
+                VALUES (
+                    (select CandID from candidate where PSCID = :pscid),
+                    :genomic_file_id
+                )
+            "
+            );
 
-            $prep   = $DB->prepare($stmt_prefix . $stmt);
-            $result = $DB->execute($prep, array(), array('nofetch' => true));
-
+            foreach ($headers as $pscid) {
+                $pscid   = trim($pscid);
+                $success = $stmt->execute(
+                    array(
+                     "pscid"           => $pscid,
+                     "genomic_file_id" => $fileToUpload->GenomicFileID,
+                    )
+                );
+                if (!$success) {
+                    throw new DatabaseException("Failed to insert row");
+                }
+            }
         } catch (DatabaseException $e) {
+            endWithFailure();
```

Comment:
@driusan This is definitely better than it was. But I agree that it could be improved (or rethink).
Lets merge